### PR TITLE
Implement full parsing logic for Add command

### LINKED ISSUE #68: Implement full parsing logic for Add command
For different combinations of options:

- add -o [owner] -p [petpatient] -a [appointment]
**To add a new owner, a new pet patient and a new appointment**

- add -a [appointment] -o [owner's nric] -p [pet's name]
**To make a new appointment for a pet patient under an existing owner**

- add -p [petpatient] -o [owner's nric]
**To add a new pet patient under an existing owner**

- add -o [owner]
**To add a new person (independent)**

#67

### PR DESCRIPTION
**Pull request by Aquarinte**
- Updated the relevant classes: Appointment, PetPatient, Person.

- Consolidated the various 'add' commands' logic and parsing into AddCommand and AddCommandParser classes. 

- Implemented full logic for combination of parsing options for Add command #68 

- Removed redundant classes: AddAppointmentCommand, AddAppointmentCommandParser, AddPetPatientCommand, AddPetPatientCommandParser.

Fixes #68

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -87,22 +140,85 @@ public AddCommand(PetPatient pet) {
...
+                }
+
+                toAddAppt.setPetPatientName(petPatientName);
+                model.addAppointment(toAddAppt);
+                return new CommandResult(String.format(message, toAddAppt, owner));
+
+            case 3: //add new pet patient under an existing person
+                Person p = getPersonOfNric();
+                if (p != null) {
+                    toAddPet.setOwnerNRIC(ownerNric);
```
Comment (chialejing) [inline src/main/java/seedu/address/logic/commands/AddCommand.java]: Checkstyle error: Should change to "setOwnerNric".

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -69,17 +78,93 @@ public Person parsePerson(String ownerInfo) throws ParseException {
...
-    public AddCommand parseNewOwnerPetAppt(String ownerInfo, String petInfo, String apptInfo)
+    public AddCommand createNewOwnerPetAppt(String ownerInfo, String petInfo, String apptInfo)
             throws ParseException {
-        System.out.println("I AM PARSING 3 NOW");
         Person owner = parsePerson(ownerInfo);
-        PetPatient petPatient = new AddPetPatientCommandParser().parse(petInfo, owner);
-        Appointment appt = new AddAppointmentCommandParser().parse(apptInfo, owner, petPatient);
+
+        PetPatient petPatient = parsePetPatient(petInfo);
+        petPatient.setOwnerNRIC(owner.getNric());
```
Comment (chialejing) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java]: Checkstyle error: Should change to "setOwnerNric".

File: src/main/java/seedu/address/model/petpatient/PetPatient.java [hunk 3]
```diff
@@ -111,8 +114,12 @@ public String getBloodType() {
         return bloodType;
     }
 
-    public Person getOwner() {
-        return owner;
+    public Nric getOwner() {
+        return ownerNric;
+    }
+
+    public void setOwnerNRIC(Nric ownerNric) {
```
Comment (chialejing) [inline src/main/java/seedu/address/model/petpatient/PetPatient.java]: Checkstyle error: Should change to "setOwnerNric".

File: src/main/java/seedu/address/model/petpatient/PetPatient.java [hunk 4]
```diff
@@ -8,7 +8,8 @@
 import java.util.Optional;
 import java.util.Set;
 
-import seedu.address.model.person.Person;
+//import seedu.address.commons.events.ui.NewResultAvailableEvent;
+import seedu.address.model.person.Nric;
```
Comment (chialejing) [inline src/main/java/seedu/address/model/petpatient/PetPatient.java]: Checkstyle error: 'seedu.address.model.person.Nric' should be separated from previous import group by one line. [CustomImportOrder]

**Comment by chialejing** [comment]
@Aquarinte I have modified to resolve all relevant Checkstyle errors. Please update your copy! :)

**Comment by Aquarinte** [comment]
Thanks! @chialejing

**Pull request merged**